### PR TITLE
10805 event preview image generator add event to dn when page is ready

### PR DESCRIPTION
**Pull request by Akiyamka**

### CODE REVIEW

File: src/core/metrics/app-metrics.ts [hunk 1]
```diff
@@ -3,25 +3,52 @@ import { Sequence } from './sequence';
...
+  private completed: Record<string, number> = {};
+  private eventLog: string[] = [];
+  private settings = new SessionSettings<'KONTUR_SQ_ALERT' | 'KONTUR_SQ_LOG'>();
 
   constructor() {
     globalThis.KONTUR_METRICS = {
       sequences: this.sequences,
       events: this.eventLog,
+      toggleAlert: () => this.settings.toggle('KONTUR_SQ_ALERT'),
+      toggleLog: () => this.settings.toggle('KONTUR_SQ_LOG'),
```
Comment (IlyaIzr) [inline src/core/metrics/app-metrics.ts:37]: Why theese needed for global window object?

**Review by IlyaIzr — APPROVED** [review]
Works as expected although the whole thing is hard to grasp

File: src/core/metrics/app-metrics.ts (hunk 1 above)
Comment (Akiyamka) [reply src/core/metrics/app-metrics.ts:37]: Mostly for testers and debugging. Testers and back-ends want to see is event work right on fe side, so I add such "debug tools". Now the can write just `KONTUR_METRICS.toggleAlert()` for test event, and `toggleLog` to collect logs for task

**Comment by Akiyamka** [comment]
@IlyaIzr Can you tell me which parts were difficult? I would revise them or at least add a commentary

**Comment by IlyaIzr** [comment]
> @IlyaIzr Can you tell me which parts were difficult? I would revise them or at least add a commentary

Makes more sense to me now when I read it second time. in `src/core/metrics/sequences` set of string events looks somewhat random, may be some naming structure there would've helped

**Comment by Akiyamka** [comment]
> > @IlyaIzr Can you tell me which parts were difficult? I would revise them or at least add a commentary
> 
> Makes more sense to me now when I read it second time. in `src/core/metrics/sequences` set of string events looks somewhat random, may be some naming structure there would've helped

agree, in can be improved by some atom naming convention -  right now they have pretty random names.
Inside reatom action name it string `${atomName}_${actionName}`

**Review by andrei-kucherov — APPROVED** [review]

**Pull request merged**